patches/chrome-browser-ui-startup-startup_browser_creator_impl.cc.patch

```diff
@@ -1,12 +1,30 @@
 diff --git a/chrome/browser/ui/startup/startup_browser_creator_impl.cc b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
-index 07dc31daf2d0b7c398b38a6146e275b55a80ebec..230c9e2b25d69696761e7a6eae6394dc16a56fdb 100644
+index 3236f6469951e0001bb14651c98c4e76a5667d00..05c89250989f30d9cebe5282a9b3f91815e165e1 100644
 --- a/chrome/browser/ui/startup/startup_browser_creator_impl.cc
 +++ b/chrome/browser/ui/startup/startup_browser_creator_impl.cc
-@@ -411,6 +411,7 @@ StartupBrowserCreatorImpl::DetermineURLsAndLaunch(
-   welcome_enabled =
-       welcome::IsEnabled(profile_) && welcome::HasModulesToShow(profile_);
- #endif  // BUILDFLAG(ENABLE_DICE_SUPPORT)
-+  BRAVE_STARTUPBROWSERCREATORIMPL_DETERMINEURLSANDLAUNCH
+@@ -529,6 +529,7 @@ StartupBrowserCreatorImpl::DetermineStartupTabs(
+       }
+ #endif
+ 
++      BRAVE_STARTUPBROWSERCREATORIMPL_DETERMINESTARTUPTABS
+       // Potentially add the What's New Page. Note that the What's New page
+       // should never be shown in the same session as any first-run onboarding
+       // tabs. It also shouldn't be shown with reset tabs that are required to
+@@ -552,7 +553,7 @@ StartupBrowserCreatorImpl::DetermineStartupTabs(
+     // Note that URLs from preferences are explicitly meant to override showing
+     // the NTP.
+     if (prefs_tabs.empty()) {
+-      AppendTabs(provider.GetNewTabPageTabs(*command_line_, profile_), &tabs);
++      if (!has_first_run_experience) AppendTabs(provider.GetNewTabPageTabs(*command_line_, profile_), &tabs);
```

Comment:
is this a wip? These patches definitely do not follow patching standards

Reply:
Fixed here 51f8840abedea20b807f9d16864f8a15af410c82 using your advice to override `GetDistributionFirstRunTabs`.